doc/sphinx-guides/source/api/native-api.rst

```diff
@@ -459,13 +459,24 @@ For example::
 Add a File to a Dataset
 ~~~~~~~~~~~~~~~~~~~~~~~
 
-Add a file to an existing Dataset. Description and tags are optional::
+When adding a file to a dataset, you can optionally specify the following:
+
+*   A description of the file.
+*   Tags and categories for the file.
+*   Terms of use and access for the file.
+
+
+Add a file to an existing Dataset. Description, terms of use and tags are optional::
 
     POST http://$SERVER/api/datasets/$id/add?key=$apiKey
 
-A more detailed "add" example using curl::
+A more detailed "add" examples using curl::
+
+    curl -H "X-Dataverse-key:$API_TOKEN" -X POST -F 'file=@data.tsv' -F 'jsonData={"description":"My description.","categories":["Data"], "termsOfUseAndAccess":{"termsType":"LICENSE_BASED", "license":"Apache Software License 2.0"}}' "https://example.dataverse.edu/api/datasets/:persistentId/add?persistentId=$PERSISTENT_ID"
+
+    curl -H "X-Dataverse-key:$API_TOKEN" -X POST -F 'file=@data.tsv' -F 'jsonData={"termsOfUseAndAccess":{"termsType":"ACADEMIC_PURPOSE_AND_NOT_FOR_REDISTRIBUTION"}}' "https://example.dataverse.edu/api/datasets/:persistentId/add?persistentId=$PERSISTENT_ID"
```

Comment:
Isn't this RestrictType not TermsOfUSeType?

Reply:
I feel like we lack a place where I can look at the list of available options. Since atm I can see the examples but I don't know how many more options are available. If we have an endpoint for that, we should mention that it return's the list that we can use to fill the fields.

Reply:
An endpoint for listing active licenses should be added in the (near) future, so maybe then we can update this docs a little more.
Also, explanation how to use this endpoint in detail (meaning: what parameters combinations should be used ect.) should be in user docs and not in here.

Reply:
> Isn't this RestrictType not TermsOfUSeType?

Those are parameters that we have agreed to. We have made them more 'user-friendly' in accordance with @wfenrich wishes.

Reply:
Ok then we need to explain what we can put there, since if I use it I wouldn't be sure what to put there, since the enums are not "compatible" with those parameters.

Reply:
I'm not sure where should it be since they are mixing stuff. For example search.rst doc contains available parameters.